### ISSUE TITLE
[Feature Request] Minor Improvements

### ISSUE THREAD

**Issue by greenDarkness** [Admiral-Fish/PokeFinder#91]
1. ~~CSV exporting for the Gen 4 Egg Generator and Researcher page.~~ (Added)
2. A "Search Calls" function for the Gen 4 Wild and Egg pages - Basically the same one from "Seed to Time" except you can filter the range of the search
3. ~~For Cherrygrove City, the fishing encounters for Good Rod/Super Rod at night are incorrect. Corsola cannot be caught at night; it's encounter should be replaced by Staryu.~~ (Added)
4. ~~Replace the male and female symbols to be M and F respectively when exporting the data. The non-ASCII characters currently used are interpreted as "?" to Excel.~~
5. ~~A check box on the time finder page that says something to the effect of "Parents in day daycare" which is the equivalence of having 2 roamers.~~ ~~Changed to: modify the call searcher to work based on longest matching substring instead of total string.~~ (Added)
6. ~~Increase the number of coin flips displayed as sometimes even 15 flips isn't enough to get only 1 result.~~ (Added)
7. Stupid proofing - if the user searches for a specific gender with the ratio set to genderless Pokefinder will run the search but there will be no results. 
8. Routes are not listed in ascending order properly. Example: Routes 5-9 are listed after route 48. This can be fixed by adding 0's to the start of each number.
~~9. Restore gen 4 wild occidentary since it's useful for Pokerus/Item rng.~~ Moved to #129 
10. Search for two-three potential targets x number of frames appart. Useful for doing all the dog/birbs at the same time. 
11. CSV/TXT exporting for the Gen 4 Seed to Time page.
12. A option in profiles to set if they hit even or odd seeds.

**Comment by Admiral-Fish** [comment]
1. Sure
2. Sure
3. Gonna have to prove it. Encounter tables are directly from the game
4. Idk about this one. The tableview just grabs data directly from the model without really knowing what information it is grabbing.
5. I don't know what this means.
6. Sure

**Comment by greenDarkness** [comment]
1. Thanks!
2. Sure. So I pm'd you a prototype that I made in excel. So how I envision this to work would be:
Generate a seed with Pokefinder
Click a "Search Calls Button"
The standard Pokefinder "Search Calls" window will appear, except it will have two boxes where one can add a frame based range.
The users can then use the "K", "E", and "P" buttons as normal to search through the range specified.
For example, if the users entered 1-100 in his range, and then clicked "K, E, E, P". Pokefinder would then check through the calls for frames 1-100 to find the calls "K, E, E, P" and then display how many matches there were.
The users would continue doing calls until their is 1 match, and then clicking "OK" would take the user to the frame of the last letter they entered of the search.
3. Sure, would a video of me catching Staryu at night prove it? Since it isn't listed as an encounter there in Pokefinder?
4. It's fine then. I only wanted 4. for 2. since I had been doing it manually.
5. Let me rephrase this one. A box in the "Search" section of "Seed to Time" that allows a user to input a number, and then that number of calls will be skipped in addition to the number of calls skipped from roamers. Example:
The user has 3 roamers checked, and enters a 1 into the proposed box.
When the user click "Generate", calls will be generated as normal except their will be 4 skips instead of 3 since the user entered a 1 into the proposed box.
6. Thanks!

**Comment by Admiral-Fish** [comment]
2. So when I first read this, I thought you mean modify Seed to Time which I thought was a bad idea. Could probably have it be a right click tableview action that opens a dialog to allow user input and handle searching through the model. Would have to be slightly different then the existing setup.

3. I did some looking around. Game data does not reference time of day impacting anything other then grass encounters. However, Staryu apparently is a swarm encounter in Cherrygrove City. Perhaps this is what you mean? Also this is only the case for Good Rod, not Super Rod. 
![image](https://user-images.githubusercontent.com/24730718/81356400-f0030000-9085-11ea-86c9-496b02a20f5a.png)

5. I don't see how this is useful.

**Comment by greenDarkness** [comment]
2. Okay, that sounds great.
3. I cannot say for sure that is was or was not a swarm encounter since I didn't check at the time. However, Serebii does say that Corsola and Staryu are morning/day and night only encounters respectively. I'll do some testing.
https://www.serebii.net/pokearth/johto/cherrygrovecity.shtml#goodrod
4. This is useful if the users is abusing in an area with NPCs, and one of the NPCs advances the frame before the users can open the menu to check. If the frame was advanced then the users won't be able to search for their seed with the "Search Calls" function. However, if skips can be manually increased than the users can still use the "Search Calls" function even if their frame was advanced.

**Comment by Admiral-Fish** [comment]
2. Will try to do this before I start focusing on gen 5
3. Fixed
4. I don't like this idea as you have to guess NPC advances. I think it would be better to modify the call searcher to work based on longest matching substring instead of total string

**Comment by greenDarkness** [comment]
2. That would be awesome. Then you'd only have to do calculations for radio toggles outside of Pokefinder.
3. Thanks! I crossed it off the list!
4. Longest matching substring is even better since it would take out the guess work.

**Comment by Admiral-Fish** [comment]
It has been forever since I have looked at this but it seems most topics were addressed. Feel free to reopen or create another issue if that is not the case.

**Comment by greenDarkness** [comment]
Sorry, didn't mean to lump them all together haha. 

I'll give them a look over in a bit!
